catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/TestIcebergTable.java

```diff
@@ -463,16 +463,19 @@ public void testAlterIcebergTable() {
     Column[] expected =
         new Column[] {
           new IcebergColumn.Builder()
+              .withId(2)
               .withName("col_2_new")
               .withType(Types.DateType.get())
               .withComment(ICEBERG_COMMENT)
               .build(),
           new IcebergColumn.Builder()
+              .withId(1)
               .withName("col_1")
               .withType(Types.IntegerType.get())
               .withComment(ICEBERG_COMMENT + "_new")
               .build(),
           new IcebergColumn.Builder()
+              .withId(3)
```

Comment:
@FANNG1, @Clearvive, IMO, comparing the id field is necessary for equality, otherwise the id of `IcebergColumn` is meaningless.

Reply:
I'm checking the overall logic of id of `IcebergColumn`

Reply:
seems we could remove `Id` from IcebergColumn since we just retrieve it from Iceberg and not used any more. @SteNicholas do you like to remove `Id` in this pr?